source/extensions/filters/network/common/redis/redis_command_stats.cc

```diff
@@ -0,0 +1,59 @@
+#include "extensions/filters/network/common/redis/redis_command_stats.h"
+
+#include "extensions/filters/network/common/redis/supported_commands.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+RedisCommandStats::RedisCommandStats(Stats::Scope& scope, const std::string& prefix,
+                                     bool enableCommandCounts)
+    : scope_(scope), stat_name_set_(scope.symbolTable()), prefix_(stat_name_set_.add(prefix)),
```

Comment:
it's probably better to just createScope from the scope, instead of adding the prefix every time.  The ScopePrefixer has a more efficient implementation there.

Reply:
`createScope` seems like the right thing to do, however when I tested it I get duplicate histogram metrics, but not counters, so I've gone back to more complex `DynamoStats`-like approach.

Reply:
That's very strange, scope is used everywhere else.

Reply:
Per slack DM, I think you should make a new Scope only if you want the lifetime of the contained stats to differ from that of the parent scope. Otherwise the extra scope will require creation of a bunch of cache entries (central cache and 1 TLS cache entry per thread) in source/common/stats/thread_local_store.cc. So my take is that we should use use the prefix string here but I'd love to hear @mattklein123's opinion.

Reply:
Yeah agree with @jmarantz. We should be able to make sure cheaper w/ symbol table stuff also which @jmarantz can guide you on.

Reply:
Ok, that makes sense, thanks